mpas_analysis/test/test_climatology.py

```diff
@@ -0,0 +1,250 @@
+                     for month in [1, 2, 3]]
+
+        variableMap = {'mld': ['timeMonthly_avg_tThreshMLD'],
+                       'Time': [['xtime_startMonthly', 'xtime_endMonthly']]}
+        variableList = ['mld']
+
+        ds = open_multifile_dataset(
+            fileNames=fileNames,
+            calendar=calendar,
+            config=config,
+            timeVariableName='Time',
+            variableList=variableList,
+            variableMap=variableMap)
+
+        assert(len(ds.Time) == 3)
+        return ds
+
+    def test_compute_climatology(self):
+        config = self.setup_config()
+        calendar = 'gregorian_noleap'
+        ds = self.open_test_ds(config, calendar)
+
+        assert('month' not in ds.coords.keys())
+        assert('daysInMonth' not in ds.coords.keys())
+
+        # test add_months_and_days_in_month
+        ds = climatology.add_months_and_days_in_month(ds, calendar)
+
+        self.assertArrayEqual(ds.month.values, [1, 2, 3])
+        self.assertArrayEqual(numpy.round(ds.daysInMonth.values), [31, 28, 31])
+
+        # test compute_climatology on a data set
+        monthNames = 'JFM'
+        monthValues = constants.monthDictionary[monthNames]
+        dsClimatology = climatology.compute_climatology(ds, monthValues,
+                                                        calendar)
+
+        assert('Time' not in dsClimatology.dims.keys())
+
+        self.assertEqual(dsClimatology.data_vars.keys(), ['mld'])
+
+        climFileName = '{}/refSeasonalClim.nc'.format(self.datadir)
+        refClimatology = xarray.open_dataset(climFileName)
+        self.assertArrayApproxEqual(dsClimatology.mld.values,
+                                    refClimatology.mld.values)
+
+        # test compute_climatology on a data array
+        mldClimatology = climatology.compute_climatology(ds.mld, monthValues,
+                                                         calendar)
+
+        assert('Time' not in mldClimatology.dims)
+
+        self.assertArrayApproxEqual(dsClimatology.mld.values,
+                                    mldClimatology.values)
+
+        # for good measure...
+        self.assertArrayApproxEqual(mldClimatology.values,
+                                    refClimatology.mld.values)
+
+    def test_compute_monthly_climatology(self):
+        config = self.setup_config()
+        calendar = 'gregorian_noleap'
+        ds = self.open_test_ds(config, calendar)
+
+        monthlyClimatology = climatology.compute_monthly_climatology(ds,
+                                                                     calendar)
+
+        assert(len(monthlyClimatology.month) == 3)
+
+        self.assertEqual(monthlyClimatology.data_vars.keys(), ['mld'])
+
+        climFileName = '{}/refMonthlyClim.nc'.format(self.datadir)
+        refClimatology = xarray.open_dataset(climFileName)
+
+        self.assertArrayApproxEqual(monthlyClimatology.mld.values,
+                                    refClimatology.mld.values)
```

Comment:
Thanks @xylar, this will be really helpful to make sure we (or others!) don't accidentally break the key computation in MPAS-Analysis!